Makefile

```
@@ -0,0 +1,13 @@
+.PHONY: test tests
+NIX_PATH := nixpkgs=https://github.com/NixOS/nixpkgs/archive/nixpkgs-unstable.tar.gz
+
+all: all-tests
+
+all-tests:
+	nix-shell --pure tests -I ${NIX_PATH} -A run.all
```

You can prevent some duplication here:

```suggestion
all-tests: TEST = all
all-tests: test
```

---

Doesn't seem to work for me:
```
$ make
Makefile:11: *** Use 'make test TEST=<test_name>' to run desired test.  Stop.
```

---

I got it to work by calling `make` recursively. WDYT?

---

Ah I hadn't tested it with an `ifndef`... But this seems fine too

---

Oh wait, you should use `$(MAKE)` instead of `make` within make recipes, as that will pass the command line flags of this call on to the recursive call.